apache-spark-master/build/Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM ubuntu
+
+MAINTAINER Frank Austin Nothaft, fnothaft@berkeley.edu
+
+RUN apt-get update && \
+  apt-get install -y \
+    python \
+    libnss3 \
+    openjdk-8-jdk \
+    curl
+
+COPY download.sh /home/
+RUN sh /home/download.sh
```

Comment:
@fnothaft has a point, but try the clone once manually to make sure that it's not considerably slower that dl'ing the tarball.